chainer/functions/connection/embed_id.py

```diff
@@ -41,6 +41,10 @@ def forward(self, inputs):
             return xp.where(
                 mask[..., None], 0, W.take(xp.where(mask, 0, x), axis=0)),
 
+        if not issubclass(type(W), type(x)):
```

Comment:
Is it enough to check `W` is a subclass of `type(x)`? I mean, should we check the case `x` is a subclass of `numpy.ndarray`?